autofaiss/indices/distributed.py

```diff
@@ -355,7 +378,315 @@ def add_embeddings_to_index(
+        embedding_ids_df_handler=write_ids_df_to_parquet_fn,
+        index_optimizer=optimize_index_fn
+    )
+
+
+def _add_embeddings_to_index_local(
+    trained_index_path: str,
+    embedding_reader: EmbeddingReader,
+    memory_available_for_adding: int,
+    make_direct_map: bool,
+    embedding_ids_df_handler: Optional[Callable[[pd.DataFrame, int], Any]],
+    index_optimizer: Optional[Callable]
+) -> Dict[str, str]:
+    """Add embeddings to small index"""
+
+    with tempfile.TemporaryDirectory() as tmp:
+        trained_index_tmp_path = os.path.join(tmp, "index")
+        trained_index = _load_index(trained_index_path, trained_index_tmp_path)
+        
+        batch_size = get_optimal_batch_size(embedding_reader.dimension, memory_available_for_adding)
+        all_ids = []
+        for (vec_batch, ids_batch) in embedding_reader(batch_size=batch_size):
+            consecutive_ids = ids_batch["i"].to_numpy()
+            trained_index.add_with_ids(vec_batch, consecutive_ids)
+            if embedding_ids_df_handler:
+                all_ids.append(ids_batch)
+
+        if embedding_ids_df_handler:
+            embedding_ids_df_handler(pd.concat(all_ids), 0)
+    
+        if make_direct_map:
+            trained_index.make_direct_map()
+
+        return index_optimizer(trained_index, "")
+
+def _add_embeddings_to_index_from_hdfs_distributed(
+    trained_index_path: str,
+    trained_index_key: str,
+    embedding_root_dir: str,
+    embedding_count: int,
+    current_memory_available: str,
+    embedding_column_name: str,
+    output_root_dir: str,
+    max_index_query_time_ms: float,
+    min_nearest_neighbors_to_retrieve: int,
+    use_gpu: bool,
+    num_cores_per_executor: Optional[int],
+    temp_root_dir: str,
+    id_columns: Optional[List[str]],
+    nb_splits_per_big_index,
+    make_direct_map: bool
+):
+    partition = extract_partition_name_from_path(embedding_root_dir)
+    partition_temp_root_dir = os.path.join(temp_root_dir, "add_embeddings", partition)
+
+    return _add_embeddings_to_index_from_hdfs(
+        add_embeddings_fn=partial(
+            add_embeddings_to_index_distributed,
+            faiss_index_or_path=trained_index_path,
+            make_direct_map=make_direct_map,
+            num_cores_per_executor=num_cores_per_executor,
+            temporary_indices_folder=partition_temp_root_dir,
+            nb_indices_to_keep=nb_splits_per_big_index,
+        ),
+        embedding_root_dir=embedding_root_dir,
+        output_root_dir=output_root_dir,
+        index_key=trained_index_key,
+        embedding_column_name=embedding_column_name,
+        current_memory_available=current_memory_available,
+        id_columns=id_columns,
+        max_index_query_time_ms=max_index_query_time_ms,
+        min_nearest_neighbors_to_retrieve=min_nearest_neighbors_to_retrieve,
+        use_gpu=use_gpu,
+        make_direct_map=make_direct_map
+    )
+
+
+def add_embeddings_to_indexes_from_hdfs(
```

Comment:
If this function works for other file system like S3, better to rename it.
For example: `add_embeddings_to_indexes_from_remote_file_system` or `add_embeddings_to_indexes_from_remote_fs`

Reply:
Done